src/Raven.Client/Documents/Linq/RavenQueryableExtensions.cs

```diff
@@ -135,5 +138,20 @@ public static bool ContainsAll<T>(this IEnumerable<T> list, IEnumerable<T> items
             throw new InvalidOperationException(
                 "This method isn't meant to be called directly, it just exists as a place holder for the LINQ provider");
         }
+
+        /// <summary>
+        /// Implementation of AsAsyncEnumerable
+        /// </summary>
+        public static async IAsyncEnumerable<TSource> AsAsyncEnumerable<TSource>(this IRavenQueryable<TSource> source)
```

Comment:
1. IRavenQueryable => IQuerable I think
2. What we want to do is to add `IAsyncEnumerable<T>` inheritance to `IRavenQueryable<T>` then check here is source implements that interface, if not, convert that to IRavenQueryable (not sure if 2nd part is needed)
3. Implement inside the RavenQueryInspector GetAsyncEnumerator method

Reply:
We cannot use StreamAsync enumerator here, because we would have discrepancy between `foreach` and `await foreach`. One would track entities, 2nd one would not.

Reply:
done.